frontend/src/components/dialogs/GameDetailsDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {Dialog, DialogContent, DialogTitle, Stack, Typography} from '@mui/material';
+import {useDialog} from './useDialog.ts';
+import {default as _ReactPlayer} from 'react-player/lazy';
+import {ReactPlayerProps} from "react-player/types/lib";
+
+const ReactPlayer = _ReactPlayer as unknown as React.FC<ReactPlayerProps>;
+
+export interface GameDetailsDialogProps {
+    title: string;
+    description: string;
+    releaseDate: string;
+    developerName: string;
+    score: string;
+    youtubeTrailer: string;
+}
+
+const GameDetailsDialog: React.FC = () => {
+    const {dialogProps, closeDialog} = useDialog();
+
+    if (!dialogProps || !(dialogProps as GameDetailsDialogProps).title) {
+        return null;
+    }
+
+    const {
+        title,
+        description,
+        releaseDate,
+        developerName,
+        score,
+        youtubeTrailer
+    } = dialogProps as GameDetailsDialogProps;
```

Comment:
Warum hier noch mal der extra TypeCast? müsste auch ohne funktionieren.

Reply:
Während ich den Code darüber erfolgreich in dialogProps?.title === null ändern konnte, bekomme ich an dieser Stelle allerdings einen Fehler, wenn ich es nicht extra type caste: 

![image](https://github.com/dforstmaierLise/gleam/assets/171693026/f3b1dd91-81fe-4cde-bc51-72bd78e14184)